mutt/file.c

```diff
@@ -1397,7 +1397,13 @@ size_t mutt_file_tidy_path(char *buf, bool rsym)
   char s[PATH_MAX];
 
   if (rsym)
-    return mutt_str_strfcpy(buf, NONULL(realpath(buf, s)), PATH_MAX);
+  {
+    char *rp = realpath(buf, s);
+    if (rp)
+      return mutt_str_strfcpy(buf, rp, PATH_MAX);
+    else
+      return mutt_str_strlen(buf);
+  }
```

Comment:
@bazzinotti 
This breaks one of your test cases.
If the path doesn't exist, `realpath()` returns `NULL` ...
Before: and the original path was killed
After: the original path is left untouched

I _think_ this is a saner thing to do.

Reply:
I think you are right. Btw nice job retouching the code :bowing_man: 
I've modified the test so that if `realpath` fails, it expects the output path to match the input path.